src/httpClient.ts

```diff
@@ -72,7 +73,8 @@ export default class HttpClient {
           error: serializeError(error),
         });
 
-        throw error;
+        const hostname = error.config?.url ? new URL(error.config.url).hostname : 'N/A';
+        throw new ClientException(hostname, error);
```

Comment:
This is something we need to discuss - the `ClientException` normally requires a service name. We could either pass it to all client calls or use the hostname like this

Reply:
Hmm, the hostname is not a bad start.

Reply:
Maybe it's even better long term than passing an arbitrary name. At least this way it shows more exactly what failed